app/controllers/admin_incoming_message_controller.rb

```diff
@@ -36,6 +36,26 @@ def destroy
     redirect_to admin_request_url(@incoming_message.info_request)
   end
 
+  def bulk_destroy
+    if params[:commit] == "No"
+      redirect_to(admin_request_url(params[:request_id]))
+    end
+
+    @incoming_messages = IncomingMessage.where(:id => params[:ids].split(","))
+    if params[:commit] == "Yes"
+      info_request = InfoRequest.find(params[:request_id])
+      @incoming_messages.each do |message|
+        message.destroy
```

Comment:
What happens if a message can't be destroyed (e.g. some weird DB constraint issue)?

I think this should be more like:

```ruby
@incoming_messages.each do |message|
  if message.destroy
    # log event
  else
    errors < message.id
  end
end

# ...

if errors.empty?
  flash[:notice] = 'Incoming messages successfully destroyed.'
else
  flash[:error] = <<-EOF.strip_heredoc
  Only some incoming messages were destroyed.
  Incoming Messages #{ errors.join(', ') } could not be destroyed.
  Try destroying them individually.
  EOF
end